src/transactions/transactionBytes.js

```diff
@@ -203,132 +148,110 @@ function getEmptyBytesForTransaction(transaction) {
+			? Buffer.from(transaction.signature, 'hex')
+			: Buffer.alloc(0);
 
-			for (let i = 0; i < 8; i++) {
-				transactionBuffer.writeByte(recipient[i] || 0);
-			}
-		} else {
-			for (let i = 0; i < 8; i++) {
-				transactionBuffer.writeByte(0);
-			}
-		}
-		transactionBuffer.writeLong(transaction.amount);
+		this.transactionSecondSignature = transaction.signSignature
+			? Buffer.from(transaction.signSignature, 'hex')
+			: Buffer.alloc(0);
 
-		if (transaction.asset.data) {
-			const dataBuffer = Buffer.from(transaction.asset.data);
-			for (let i = 0; i < dataBuffer.length; i++) {
-				transactionBuffer.writeByte(dataBuffer[i]);
-			}
-		}
-
-		if (assetSize > 0) {
-			for (let i = 0; i < assetSize; i++) {
-				transactionBuffer.writeByte(assetBytes[i]);
-			}
-		}
 
-		if (transaction.signature) {
-			assignHexToTransactionBytes(transactionBuffer, transaction.signature);
-		}
+		this.checkTransaction();
+		return this;
+	}
 
-		if (transaction.signSignature) {
-			assignHexToTransactionBytes(transactionBuffer, transaction.signSignature);
-		}
+	/**
+	 * @method get transactionBytes
+	 * @return {Buffer}
+	 */
 
-		transactionBuffer.flip();
-		const arrayBuffer = new Uint8Array(transactionBuffer.toArrayBuffer());
-		const buffer = [];
+	get transactionBytes() {
+		return this.concatTransactionBytes();
+	}
 
-		for (let i = 0; i < arrayBuffer.length; i++) {
-			buffer[i] = arrayBuffer[i];
-		}
+	/**
+	 * @method concatTransactionBytes
+	 * @private
+	 * @return {Buffer}
+	 */
 
-		return Buffer.from(buffer);
+	concatTransactionBytes() {
+		return Buffer.concat([
+			this.transactionType,
+			this.transactionTimestamp,
+			this.transactionSenderPublicKey,
+			this.transactionRequesterPublicKey,
+			this.transactionRecipientID,
+			this.transactionAmount,
+			this.transactionAssetData,
+			this.transactionSignature,
+			this.transactionSecondSignature,
+		]);
 	}
 
-	// Get Transaction Size and Bytes
-	const transactionAssetSizeBuffer = getEmptyBytesForTransaction(transaction);
-	const assetSize = transactionAssetSizeBuffer.assetSize;
-	const assetBytes = transactionAssetSizeBuffer.assetBytes;
-
-	const emptyTransactionBuffer = createEmptyTransactionBuffer(assetSize);
-	const assignedTransactionBuffer = assignTransactionBuffer(
-		emptyTransactionBuffer, assetSize, assetBytes,
-	);
+	/**
+	 * @method getAssetBytes
+	 * @private
+	 * @return {Buffer}
+	 */
 
-	return assignedTransactionBuffer;
-}
+	getAssetBytes() {
+		return getAssetBytesHelper(this.transaction);
+	}
 
-/**
- * @method getBytes
- * @param transaction Object
- *
- * @return {buffer}
- */
+	/**
+	 * @method checkTransaction
+	 * @throws
+	 * @return {}
+	 */
 
-function getTransactionBytes(transaction) {
-	return createTransactionBuffer(transaction);
+	checkTransaction() {
+		if (this.transaction.type === 0 && this.transaction.asset.data) {
+			if (this.transaction.asset.data.length > byteSizes.DATA
+				|| this.transactionAssetData.length > byteSizes.DATA) {
```

Comment:
If `this.transactionAssetData.length` somehow ends up less than or equal to `byteSizes.DATA` does it matter if `this.transaction.asset.data.length > byteSizes.DATA`? If the transaction ends up with `this.transactionAssetData`, does it matter if it's not true that `this.transaction.type === 0 && this.transaction.asset.data`?

Alternatively, a check for `this.transaction.asset.data.length` could be performed at the start of the constructor, rather than after all the redundant processing.